crates/video/src/vhu_video_thread.rs

```diff
@@ -0,0 +1,697 @@
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use log::{debug, warn};
+use std::{
+    fs::File,
+    mem::size_of,
+    ops::Deref,
+    os::unix::io::{AsRawFd, FromRawFd},
+    sync::{Arc, RwLock},
+};
+
+use crate::{
+    stream,
+    vhu_video::{self, Result, VideoDescriptorChain, VuVideoBackend, VuVideoError},
+    video::{self, ToBytes},
+    video_backends::VideoBackend,
+};
+use futures::executor::{ThreadPool, ThreadPoolBuilder};
+use vhost_user_backend::{VringEpollHandler, VringRwLock, VringT};
+use virtio_queue::{Descriptor, QueueOwnedT};
+use vm_memory::{
+    ByteValued, Bytes, GuestAddress, GuestAddressSpace, GuestMemory, GuestMemoryAtomic,
+    GuestMemoryMmap,
+};
+use vmm_sys_util::epoll::EventSet;
+
+type ArcVhostBknd = Arc<RwLock<VuVideoBackend>>;
+
+const MAX_BUFFERS: usize = 32;
+
+#[derive(Copy, Clone, Debug)]
+pub struct TriggeredEvent {
+    pub have_read: bool,
+    pub have_write: bool,
+    pub have_event: bool,
+    pub data: u64,
+}
+
+pub enum EventType {
+    None = 0,
+    Read,
+    Write,
+    ReadWrite,
+    Event,
+    EventRead,
+    All,
+}
+
+impl From<EventType> for epoll::Events {
+    fn from(et: EventType) -> epoll::Events {
+        match et {
+            EventType::None => epoll::Events::EPOLLERR,
+            EventType::Read => epoll::Events::EPOLLIN,
+            EventType::Write => epoll::Events::EPOLLOUT,
+            EventType::ReadWrite => epoll::Events::EPOLLIN | epoll::Events::EPOLLOUT,
+            EventType::Event => epoll::Events::EPOLLPRI,
+            EventType::EventRead => epoll::Events::EPOLLPRI | epoll::Events::EPOLLIN,
+            EventType::All => {
+                epoll::Events::EPOLLPRI | epoll::Events::EPOLLIN | epoll::Events::EPOLLOUT
+            }
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+pub struct PollerEvent {
+    pub event: epoll::Event,
+}
+
+impl PollerEvent {
+    pub fn new(et: EventType, token: u64) -> Self {
+        Self {
+            event: epoll::Event::new(epoll::Events::from(et), token),
+        }
+    }
+
+    pub fn empty() -> Self {
+        Self::new(EventType::None, 0)
+    }
```

Comment:
Should this be just the `Default` implementation or ?

Reply:
Yeah, `epoll::Events` has an `empty` method similar to this one, so I was mimicking this for `PollerEvent`. Nonetheless, it fits a `default` implementation all the same, so I will change it.